addons/mail/models/mail_tracking_value.py

```diff
@@ -119,9 +119,10 @@ def _create_tracking_values(self, initial_value, new_value, col_name, col_info,
                 'new_value_char': new_value.display_name if new_value else ''
             })
         elif col_info['type'] in {'one2many', 'many2many'}:
+            model_name = self.env['ir.model']._get(field.model).display_name
```

Comment:
I think you want field.relation

field.model is where the field is defined it seems.
I guess it's not caught in the test because the parent and child are the same model.